src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/AbstractSpeculationAnalyzer.cs

```diff
@@ -581,11 +595,22 @@ private bool ReplacementChangesSemanticsForNode(SyntaxNode currentOriginalNode,
             else if (currentOriginalNode is TExpressionSyntax originalExpression)
             {
                 var newExpression = (TExpressionSyntax)currentReplacedNode;
-                if (!ImplicitConversionsAreCompatible(originalExpression, newExpression) ||
-                    ReplacementIntroducesErrorType(originalExpression, newExpression))
-                {
+                if (!ImplicitConversionsAreCompatible(originalExpression, newExpression))
+                    return true;
+                RoslynDebug.AssertNotNull(originalExpression);
+                Debug.Assert(this.SemanticRootOfOriginalExpression.DescendantNodesAndSelf().Contains(originalExpression));
+                RoslynDebug.AssertNotNull(newExpression);
+                Debug.Assert(this.SemanticRootOfReplacedExpression.DescendantNodesAndSelf().Contains(newExpression));
+
+                var originalTypeInfo = this.OriginalSemanticModel.GetTypeInfo(originalExpression);
+                var newTypeInfo = this.SpeculativeSemanticModel.GetTypeInfo(newExpression);
+
+                // If we didn't have an error before, but now we got one, that's bad and should block conversion in all cases.
+                if (newTypeInfo.Type.IsErrorType() && !originalTypeInfo.Type.IsErrorType())
+                    return true;
```

Comment:
this is inlining the error check from before.  it is *not* something that should be overridable.